### DISCUSSION TITLE
Does this method works on complex network?

### DISCUSSION THREAD

**Discussion by 2tshhh** [GerryDazoo/Slinger#555] [Q&A]
Hello,

We are ISP and we have some slingbox all over the cities in our country! Can we re-use them based on your awesome work that you have shared with us? Be aware that we have tons of routers and switches and advance network configurations between slingboxes and the server that i'm trying to run. Should this be an issue?

Please forgive me for starting a new discussion, but i didn't find any answer about this particular question

**Comment by GerryDazoo** [comment]
Should not be a problem. Please review the networking guide PDF.

**Reply by 2tshhh** [reply]
I'm trying to run the slingbox_server.py but i get this error

Can't find specified config file config.ini Ignoring
ERROR. Cannot scan local network for slingboxes
Because the "netifaces" python modules has not been installed
Broadcast Responder Running

config.ini is in the same location with the slingbox_server.py file

**Reply by GerryDazoo** [reply]
Please read the python installation info in the readme. The missing netifaces module is not a show stopper. Also in the future please post ALL the server output.
If the config.ini file was in the same folder you wouldn't be getting this error. Please post the full command line you are using to run the server and a copy of the config.ini.

**Reply by 2tshhh** [reply]
Hello Gerry,

Currently i'm running this config for one of my 350 which is outside of my server's network. But still unsuccessfully to get the stream on VLC. However, when i try on browser http://MY-SLING-SERVER-IP:8080/remote , it opens the remote!
I'm stuck here and now after reading some posts on the discussions that you may brick your slingbox if not properly forword/block some ports, i'm afraid to test and try further experiments!  

[SLINGBOX]
sbtype="350/500/M1/M2"
;sbtype="Solo/Pro/ProHD"
;sbtype="240"
password=MY-ADMIN-PASSWORD

;uncomment the next lines ipaddress and port with your local network info
;if you don't want the server to automatically find your 
;slingbox on the local network. Needed if server and slingbox are
;not on the same LAN segment. Also removes requirement for the netifaces module  
ipaddress=SLINGBOX-IP
port=5201

;Replace with your finderid, If and only if you need remote access and don't  
;have a static ip and you plan on using my service to access you server remotely. 
;Please read the release notes. Most people don't need this.
finderid=MY-FINDER-ID

;If you've got more than one slingbox set name. THe server will use this when generating 
;logs to make it easier to see what's going on. 
;name=something

;Following are the default values
;Default resolution when server starts. You can change this and reconnect without having to restart the server
;valid range = 1..15
Resolution=12

;Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
FrameRate=30

;Min 50 Max 8000
VideoBandwidth=2000

;Min 0 (auto) max 63
VideoSmoothness=63

;Send Iframe every n seconds Max 30
IframeRate=5

;Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

;Video Source  0, 1, 2, 3 depending on your hardware corresponds to one of
;Composite, Component, S-Video, HDMI or Tuner.
;Pro:        0=(Analog)Tuner   1=Composite   2=S-Video     3=Component (Via HDMI to Component Cable)
;ProHD:      0=Tuner           1=Composite   2=S-Video     3=Component
;Solo:       0=Composite       1=S-Video     2=Component
;500:        0=Composite       1=Component   2=HDMI
;350/M1/M2:  0=Composite       1=Component
; If you don't set this, the code will use the last configured input
; I recommend setting this value. If not set correctly, you'll often only see a black screen.
; Note: Setting invalid options may cause unexpected hardware behavior. For example, setting
;  2 or 3 for M2 hardware causes it to reboot.
VideoSource = 1

[SERVER]
;local port number for the server to listen on for connections
port=8080
;maxstreams=10
;enableremote=yes

[REMOTE]
;see release notes for changing this if your remote doesn't work, But as a start make it 
;the same as your configured VideoSource 0-3. See above.
code=1
;path to an external remote control definitions file, if the defaults
;aren't working for you or you want to change the format of the web page.
;use the supplied remote.txt as a starting point.
;include=remote.txt 

Version : 4.01 Running on Windows-10-10.0.19045-SP0 pid= 13884 
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Building page for Slingbox
Using built in default remote page definition.
Starting Streamer Thread for Slingbox
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
8080 Checking for slingbox at ('SLINGBOX-IP', 5201) 0
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
Broadcast Responder Running
Server IP address is OTHER-NIC-IP
Server IP address is MY-SLING-SERVER-IP
8080 Error connecting to  ('SLINGBOX-IP', 5201)
8080 No valid slingbox ip info found in config.ini
8080 Finding Slingboxes on local network. My IP Info =  OTHER-NIC-IP
8080 Finding Slingboxes on local network. My IP Info =  MY-SLING-SERVER-IP
8080 Can't find a slingbox on network. Please make sure it's plugged in and connected. Check config.ini
8080 Giving up. Sorry...

**and at this point when i try on VLC http://MY-SLING-SERVER-IP:8080/slingbox**

03/14/2023, 10:49:45.308   Streaming request from ('MY-SLING-SERVER-IP, 57230)  /slingbox
STREAM=MY-SLING-SERVER-IP:57230:0

**when i stop VLC**

Hacker Alert. Invalid Request from  ('MY-SLING-SERVER-IP', 57247)

**Reply by GerryDazoo** [reply]
From the logs it's apparent you don't have the proper IP address and/or port number for the remote Slingbox  and/or the remote router doesn't have the correct port forwarding rule in place to allow remote access to the slingbox.

**Reply by 2tshhh** [reply]
Hello,

In order to simplify this connection, at least to run one slingbox in my life, i have order to get them physically here, and run it from LAN
So, right know my setup is, 1 router with LAN 1 my laptop and LAN 2 my Slingbox 350

When i use it on VLC http://192.168.0.14:8080/slingbox i get nothing! 

I tried Firewall protection to HIGH (as an router option, see the first pic) 
![image](https://user-images.githubusercontent.com/121289020/225916676-3b851ebb-01a5-43ba-9c1a-6eb517eb104a.png)

as well as some other Port filtering 80, 443, 3074, 3490, 3724, 5680, 8080

i tried without Port filtering and firewall turning OFF, and still no stream on vlc.

config file and server logs are below!


[SLINGBOX]
sbtype="350/500/M1/M2"
password=my password
ipaddress=192.168.0.15
port=5201
;finderid=my finder ID
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
IframeRate=5
VideoSource = 0
[SERVER]
;local port number for the server to listen on for connections
port=8080
;maxstreams=10
;enableremote=yes

[REMOTE]
;see release notes for changing this if your remote doesn't work, But as a start make it 
;the same as your configured VideoSource 0-3. See above.
code=1
;path to an external remote control definitions file, if the defaults
;aren't working for you or you want to change the format of the web page.
;use the supplied remote.txt as a starting point.
;include=remote.txt



Version : 4.01 Running on Windows-10-10.0.17763-SP0 
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Building page for Slingbox
Using built in default remote page definition.
Starting Streamer Thread for Slingbox
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
8080 Checking for slingbox at ('192.168.0.15', 5201) 0
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
Broadcast Responder Running
Server IP address is 192.168.0.14
8080 Error connecting to  ('192.168.0.15', 5201)
8080 No valid slingbox ip info found in config.ini
8080 Finding Slingboxes on local network. My IP Info =  192.168.0.14
8080 Found at 192.168.0.15 5220 " Gjilan " FinderID my finder ID ProductID 350
8080 Found matching IP Address 192.168.0.15 will use this box, check port number in your config file
8080 Using slingbox at  ('192.168.0.15', 5220)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
03/17/2023, 14:08:54.486   Streaming request from ('192.168.0.14', 58747)  /slingbox
STREAM=192.168.0.14:58747:0
8080 Using encrypted password: my password

Slinginfo  350 12 30 192.168.0.15 5220 3072 10 usxLjD0XhN84GCu
8080 Starting Stream for  192.168.0.14:58747
8080 Opening Control stream 0x2000 0 0
Connecting... ('192.168.0.15', 5220) Control
8080 Selecting Video Source 0
8080 Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 21, 24, 25, 26, 27, 28, 29, 33, 35, 36, 38, 39, 40, 41, 42, 43, 44, 46, 62, 63, 64, 65, 66, 67, 68, 69, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.0.15', 5220) Stream
8080 Stream started at 03/17/2023, 14:09:01.685  744 0
03/17/2023, 14:09:01.731  8080 Stream Terminated for  192.168.0.14:58747
8080 Shutting down connections
8080 Logging Out

**Reply by GerryDazoo** [reply]
Because you're on the LAN with the Slingbox it doesn't matter how the router is configured for now. 
You've got the wrong port number in your config.ini for the slingbox but the server software is finding a slingbox to use.
Delete the code=1 line in the remote section it's not being use because you've overridden the entry with the videosource=0 line. 
It looks like you have a problem with VLC shutting down the connection.

Please try a different media player. Like https://github.com/GerryDazoo/SlingerPlayer

**Reply by 2tshhh** [reply]
Ok, great. For the first time it worked, using the SlingPlayer
Now, i have figured out what to do on WAN next test.

However, i have just 3 question now: 
-How to know the ports of every slingbox? on config as you can see i declare 5201, but on LAN it says that the port of 2 of slingboxes i got here, are 5220 and 5221! How to know the ports of every slingbox on WAN?!
-Remote isn't working. Should i upload some kind of file of STB or something?
-The next question is, how can i configure 20 slingbox on a single config.ini ? I mean how can i switch between slingboxes on SlingPlayer?!

Thank you for your support

**Reply by GerryDazoo** [reply]
In order.
Do you mean my SlingerPlayer? not SlingPLayer?
1) You need local LAN access where the Slingboxes are located to get their port number. 5201 is the default.
2) We need a better description of what "not working" means. Can you bring the remote page up in your browser? You need a remote definition file that matches what was used when the video source was configured on the Slingbox. 
3) use the "unified_config.ini" as a template for your 20 slingboxes. If your using Windows SlingerPLayer you can type the URL into the URL combobox or click on the drop down arrow to to see all the configured Slingboxes.

**Reply by 2tshhh** [reply]
Hey Gerry,

I used your suggested SlingPLayer for Windows https://github.com/GerryDazoo/SlingerPlayer 

1. Ok, i was thinking about that. I guess i have to go on every SlingBox, in every city and connect my laptop so i can get their ports! 
2. It doesn't react on clicking the those virtual 'buttons'. I was able to bring the remote page up in my browser, as well as on your Windows SlingPlayer. 
3. I will try it.
Thank you man

**Reply by GerryDazoo** [reply]
SlingER Player.
1) At some point you'll need access to the Routers at the slingbox sites to setup the port forwarding rules to allow remote access to the slingbox.   Do you have VPN access to the remote sites? If so that's all you need to get the port numbers.
2) Please post the server output when you are clicking the buttons. Note, it only works while you're viewing a stream. As a minimum the channel digits and the power button should work. Those key codes are pretty standard.

**Reply by 2tshhh** [reply]
Hello Gerry,
I just managed to get the port of one of my slingboxes on WAN and it is working perfectly. Also remote is working as well.

Regarding to unified_config. This is how configured:

[SLINGBOXES]
peja=350
gjilani=350
kamenica=350

[350]
sbtype="350/500/M1"
password=password
;finderid=finder id
ipaddress=some ip
port=5221
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
;Remote=500_remote.txt

[350]
sbtype="350/500/M1"
password=passowrd
;finderid=finder id
ipaddress=some ip
port=5220
Resolution=5
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=2
RemoteLock=yes
;StartChannel=155
;Remote=solo_remote.txt

[350]
sbtype="350/500/M1"
password=password 
;finderid=finder id
ipaddress=some ip
port=5221
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
;Remote=m1_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=65432
maxremotestreams=2
URLbase=Gerry
enableremote=yes

But i can't see them on dropdown arrow of url's ? Should i leave them as default "sb1, sb2, sb3" ? I put the names of the location where slingboxes are.

Another question - i have set as a server one PC, but i need slingerplayer to connect from another PC, using the ip of server http://server-ip:8080/slingbox . I tested it but it doesn't connect. I checked server logs, but it seems it doesn't even get a request from other PC SlingerPlayer!

**Reply by GerryDazoo** [reply]
Did you read the release notes about the unified config? V3.08_release_notes.pdf.
You have two major problems because you didn't follow the template. Make these changes
[SLINGBOXES]
sb1=peja
sb2=gjilani
sb3=kamenica
Then replace the [350]s with
[peja]
[gjilani]
[kamenica]

then connect with http://server-ip:8080/Gerry/peja for example
Please change your URL base to something else unless you really like typing my name every time you connect.

**Reply by 2tshhh** [reply]
Gerry,

Thank You man. I managed to connect most of the Slingboxes that we got.

Thank you. You're awesome. If in the future happen to visit Kosovo, please write me

Br,
